File: chat-service/src/main/java/com/wesell/chatservice/dto/response/ChatRoomListResponseDto.java
package com.wesell.chatservice.dto.response;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.util.List;

@Getter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ChatRoomListResponseDto {
    private List<ChatRoomResponseDto> roomList;
    private boolean hasNext;
}

File: front-server/src/apis/response/chat/chat-room-list.response.dto.ts
import ResponseDto from '../response.dto';
import ChatRoomResponseDto from './chat-room.response.dto';

// eslint-disable-next-line @typescript-eslint/no-empty-interface
export default interface ChatRoomListResponseDto extends ResponseDto {
  content: {
    roomList: ChatRoomResponseDto[];
    hasNext: boolean;
  };
}
